### PR TITLE
Fix phpunit

### PR DESCRIPTION
**Pull request by wbloszyk**
This PR will allow run phpunit test by:
```
vendor/bin/simple-phpunit
bin/qa_client_ci.sh
```

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -87,7 +87,7 @@
         "behat/mink-selenium2-driver": "^1.3.1",
         "phing/phing": "^2.16",
         "symfony/dotenv": "^4.0",
-        "symfony/phpunit-bridge": "^4.0",
+        "symfony/phpunit-bridge": "^5.0",
```
Comment (jordisala1991) [inline composer.json:90]: 5.1

**Comment by wbloszyk** [comment]
Probably fixed TODOS before merging is impossible. They depends from other sonata bundles. I can remove them but this will suspended tests. I think better keep it working becouse they show deprecations and bugs (like this TODOS) which should be fixed.

**Comment by jordisala1991** [comment]
Better to have something working

File: tests/Sonata/Bundle/QABundle/CommandTestCase.php [hunk 2]
```diff
@@ -11,10 +11,10 @@
  * file that was distributed with this source code.
  */
 
-namespace Sonata\Bundle\QABundle\Tests;
+namespace Tests\Sonata\Bundle\QABundle;
```
Comment (core23) [inline tests/Sonata/Bundle/QABundle/CommandTestCase.php:14]: Would prefer `Sonata\Tests` over `Tests\Sonata` because the vendor is Sonata

Comment (wbloszyk) [reply tests/Sonata/Bundle/QABundle/CommandTestCase.php:14]: I want create sandbox as predefine application, ready to use where people can modify them. I replace tests becouse it is default directory for it. I think will be nice use `Tests\Sonata` without `Bundle\QABundle`, mybe even split it to seperate namespace like `Tests\Sonata\News`. I prefer using `Test\Sonata` becouse people can delete them without change in composer.json (this change require use `composer update`). 
```json
"autoload-dev": {
    "psr-4": {
        "Tests\\": "tests/"
    }
},
```

Comment (wbloszyk) [reply tests/Sonata/Bundle/QABundle/CommandTestCase.php:14]: BTW, should we merge QABundle with DemoBundle?

Comment (VincentLanglet) [reply tests/Sonata/Bundle/QABundle/CommandTestCase.php:14]: @core23 @wbloszyk

You can do
```
"autoload-dev": {
    "psr-4": {
        "Sonata\\Tests\\": "tests/"
    }
},
```

and then `tests/Bundle/QABundle/...`, without the `Sonata` folder.

isn't it ?

Comment (wbloszyk) [reply tests/Sonata/Bundle/QABundle/CommandTestCase.php:14]: I want avoid modify composer.json becouse if people delete tests then they have to remove it from composer.json too and run `composer update` to clear it on vendor/composer files.

Comment (jordisala1991) [reply tests/Sonata/Bundle/QABundle/CommandTestCase.php:14]: IMO having standard composer .json namespaces is better.

Comment (wbloszyk) [reply tests/Sonata/Bundle/QABundle/CommandTestCase.php:14]: IMO files from vendor directory should be trait as sonata files. Files from other directory should be trait as people files to keep it as easy is possible to edit. I dont think user will want edit composer after remove files.

@VincentLanglet What when someone add own tests? Then namespace `Sonata\Tests` will be require.

@VincentLanglet @core23 @jordisala1991 WDYT?
My sugestion is:
1. move AppBundle to src (App)
1. move QABundle to DemoBundle
1. split DemoBundle to DemoBundle and sonata-project instalator packege/bundle
1. improve DemoBundle

File: tests/Sonata/Bundle/QABundle/CommandTestCase.php [hunk 3]
```diff
@@ -72,7 +72,8 @@ public static function getAdminList()
         return [
             ['sonata.user.admin.user',                   'AppBundle\\Entity\\User\\User'],
             ['sonata.user.admin.group',                  'AppBundle\\Entity\\User\\Group'],
-            ['sonata.page.admin.page',                   'AppBundle\\Entity\\Page\\Page'],
+            // TODO: find why `sonata:admin:explain sonata.page.admin.page` suspended tests
```
Comment (jordisala1991) [inline tests/Sonata/Bundle/QABundle/CommandTestCase.php:75]: Maybe it has to do with: https://github.com/sonata-project/SonataAdminBundle/pull/6148

Comment (wbloszyk) [reply tests/Sonata/Bundle/QABundle/CommandTestCase.php:75]: i remove code from `Sonata\PageBundle\Admin\PageAdmin` and test start working:
```php
$formMapper->setHelps([
        'name' => 'help_page_name',
    ]);
```

Comment (jordisala1991) [reply tests/Sonata/Bundle/QABundle/CommandTestCase.php:75]: Then it is related to that PR, because it fixes the `setHelps`

Comment (jordisala1991) [reply tests/Sonata/Bundle/QABundle/CommandTestCase.php:75]: Can you test if it works now?

Comment (wbloszyk) [reply tests/Sonata/Bundle/QABundle/CommandTestCase.php:75]: `bin/console sonata:admin:explain sonata.page.admin.page` works correct now.

Comment (jordisala1991) [reply tests/Sonata/Bundle/QABundle/CommandTestCase.php:75]: Then, you can uncomment those, right?

Comment (wbloszyk) [reply tests/Sonata/Bundle/QABundle/CommandTestCase.php:75]: Yes, now i checking:
```
 Attempted to call an undefined method named "getType" of class "phpDocument  
!!    or\Reflection\DocBlock\Tags\InvalidTag".
```
After that this PR should be RTM.

Comment (jordisala1991) [reply tests/Sonata/Bundle/QABundle/CommandTestCase.php:75]: Do you have more stack trace of that error? To help you finding where is the wrong code/call

Comment (wbloszyk) [reply tests/Sonata/Bundle/QABundle/CommandTestCase.php:75]:
```
bin/console sonata:admin:explain sonata.customer.admin.address
```
generate that. Probably something in ecommerce generate that.

Comment (jordisala1991) [reply tests/Sonata/Bundle/QABundle/CommandTestCase.php:75]: Can you try removing this line: https://github.com/sonata-project/ecommerce/blob/3.x/src/CustomerBundle/Entity/BaseCustomer.php#L144

It does not look right

Comment (wbloszyk) [reply tests/Sonata/Bundle/QABundle/CommandTestCase.php:75]: I found it. 
https://github.com/sonata-project/ecommerce/commit/122916ba3c322b311eae1f22fef470661a48738b#diff-ae5cda693db1e676bdf271653c88e47dL91-L93
I think I can ignore it. Next ecommerce release is require.

**Comment by wbloszyk** [comment]
RTM

File: tests/Sonata/Bundle/QABundle/CommandTestCase.php (hunk 3 above)
Comment (VincentLanglet) [reply tests/Sonata/Bundle/QABundle/CommandTestCase.php]: Do you want a new ecommerce release @wbloszyk ?

Comment (wbloszyk) [reply tests/Sonata/Bundle/QABundle/CommandTestCase.php]: Will be awsome. Some error from phpDocumentor are displayed in console now.

Comment (jordisala1991) [reply tests/Sonata/Bundle/QABundle/CommandTestCase.php]: oh so it is not on 3.x, then we need a release to fix it I guess

Comment (VincentLanglet) [reply tests/Sonata/Bundle/QABundle/CommandTestCase.php]: You made this change on ecommerce @wbloszyk
https://github.com/sonata-project/ecommerce/commit/9c2a75ce27e2070eeb75349f2d542e49a0368756

This is similar to
https://github.com/sonata-project/SonataPageBundle/commit/331446bad6766aded5b6510f7bea2ec8740a3233

Witch introduced this issue https://github.com/sonata-project/SonataPageBundle/issues/1163

Did you work on the issue ? Will it introduce the same issue ? Should we revert the commit on ecommerce before releasing a new version ?

Comment (jordisala1991) [reply tests/Sonata/Bundle/QABundle/CommandTestCase.php]: Imo the cascade part is a bc break and should be reverted, it is changing the database schema

Comment (wbloszyk) [reply tests/Sonata/Bundle/QABundle/CommandTestCase.php]: @VincentLanglet @jordisala1991 I working on it. I want add new configuration parameter to all sonata bundles where people can declare which entity manager(s) should work with sonata. By default `doctrine.orm.entity_manager`. From this service it is possibe to get sql platform and prepare schema/queries for that platform.

Comment (VincentLanglet) [reply tests/Sonata/Bundle/QABundle/CommandTestCase.php]: Then we should first revert the change in order to allow a new ecommerce release.

We should even revert the change on SonataPageBundle to fix the issue if there is no other quick solution.

Comment (jordisala1991) [reply tests/Sonata/Bundle/QABundle/CommandTestCase.php]: Agree with @VincentLanglet . The first step is to revert the problematic changes, then we can find a better solution

Comment (wbloszyk) [reply tests/Sonata/Bundle/QABundle/CommandTestCase.php]: Agree:
https://github.com/sonata-project/ecommerce/pull/670/files
https://github.com/sonata-project/SonataClassificationBundle/pull/530/files
https://github.com/sonata-project/SonataPageBundle/pull/1153/files

Comment (VincentLanglet) [reply tests/Sonata/Bundle/QABundle/CommandTestCase.php]: Do you want to make the revert @wbloszyk ?
Then we'll approve and merge the PR and make new releases.

Comment (wbloszyk) [reply tests/Sonata/Bundle/QABundle/CommandTestCase.php]: Done

**Comment by wbloszyk** [comment]
@core23 @VincentLanglet Can you recheck it?

File: tests/Sonata/Bundle/QABundle/CommandTestCase.php (hunk 2 above)
Comment (core23) [reply tests/Sonata/Bundle/QABundle/CommandTestCase.php:14]: > [...] if people delete tests then they have to remove it from composer.json too and run `composer update` to clear it on vendor/composer files.

This is a sandbox project. We should risk this and follow vendor-prefixed namespaces (`Sonata\Tests`).

**Comment by jordisala1991** [comment]
Thank you @wbloszyk

**Pull request merged**